ibis/backends/flink/tests/conftest.py

```diff
@@ -19,11 +19,46 @@ def connect(*, tmpdir, worker_id, **kw: Any):
         comply with the assumption that the tests under ibis/ibis/backends/tests/
         are for batch (storage or processing) backends.
         """
-        from pyflink.table import EnvironmentSettings, TableEnvironment
-
-        env_settings = EnvironmentSettings.in_batch_mode()
-        table_env = TableEnvironment.create(env_settings)
-        return ibis.flink.connect(table_env, **kw)
+        import os
+
+        from pyflink.java_gateway import get_gateway
+        from pyflink.table import (
+            EnvironmentSettings,
+            StreamTableEnvironment,
+            TableEnvironment,
+        )
+        from pyflink.table.table_environment import StreamExecutionEnvironment
+
+        if not bool(os.environ.get("FLINK_REMOTE_CLUSTER", False)):
```

Comment:
Can we always use the remote cluster? I'd like to avoid having two options for testing here. I think we need a flink docker compose setup for a remote cluster if we don't already have one.

Reply:
I have no objection with your proposal. If we decide to go that way, I prefer to merge this in first and use a separate PR for that purpose, does that work?

Reply:
Makes sense to follow up with a PR that moves to a single way of testing.